Refatorar o código para incluir o código em uma main

Ao testar o pylint, é indicado os seguintes erros:

motion_detection.py:8:0: C0103: Constant name "is_recording" doesn't conform to UPPER_CASE naming style (invalid-name)
motion_detection.py:9:0: C0103: Constant name "recording_start_time" doesn't conform to UPPER_CASE naming style (invalid-name)
motion_detection.py:10:0: C0103: Constant name "video_writer" doesn't conform to UPPER_CASE naming style (invalid-name)

Isso ocorre por conta dessas variáveis estarem fora do loop principal do programa. Assim, o pylint as identifica como constantes globais e exigem que sejam nomeadas com letras maiúsculas.
Para corrigir, o desenvolvedor deve incluir o loop principal em uma função main

--- a/motion_detection.py
+++ b/motion_detection.py
@@ -2,86 +2,88 @@ import time
 import cv2
 
 # Parâmetros para a gravação
-# Aqui você pode ajustar conforme necessário
 MIN_CONTOUR_AREA = 15  # Área mínima do contorno para considerar como movimento
 VIDEO_DURATION = 17  # Tempo de vídeo que será gravado após detectar o movimento
 
-# Defina a captura de vídeo
-cap = cv2.VideoCapture(0)  # 0 é geralmente a câmera padrão
+def initialize_video_capture():
+    return cv2.VideoCapture(0)  # 0 é geralmente a câmera padrão
 
-# Variáveis para controle de gravação
-is_recording = False
-recording_start_time = 0
-video_writer = None
-prev_frame = None
-
-while True:
-    ret, frame = cap.read()
-    if not ret:
-        break
-    # Converter o frame para escala de cinza
+def process_frame(frame):
     gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
     gray = cv2.GaussianBlur(gray, (21, 21), 0)
+    return gray
 
-    # Inicializar o primeiro frame
-    if prev_frame is None:
-        prev_frame = gray
-        continue
-
-    # Calcular a diferença entre os frames
-    frame_delta = cv2.absdiff(prev_frame, gray)
+def detect_movement(prev_frame, gray_frame):
+    frame_delta = cv2.absdiff(prev_frame, gray_frame)
     thresh = cv2.threshold(frame_delta, 25, 255, cv2.THRESH_BINARY)[1]
-
-    # Encontrar contornos no frame de threshold
     contours, _ = cv2.findContours(thresh.copy(), cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
+    return contours
 
+def draw_contours(frame, contours):
     movement_detected = False
-
-    # Loop sobre os contornos encontrados
     for contour in contours:
         if cv2.contourArea(contour) > MIN_CONTOUR_AREA:
             movement_detected = True
-            # Calcular a caixa delimitadora mínima ao redor do contorno
             (x, y, w, h) = cv2.boundingRect(contour)
-
             cv2.rectangle(frame, (x, y), (x + w, y + h), (0, 255, 0), 2)
+    return movement_detected
 
-    if movement_detected:
-        if not is_recording:
-            # Começar a gravar
-            timestamp = time.strftime("%Y%m%d_%H%M%S")
-            video_filename = f"movement_{timestamp}.avi"
-            fourcc = cv2.VideoWriter_fourcc(*'XVID')
-            video_writer = cv2.VideoWriter(
-                video_filename, fourcc, 20.0, (int(cap.get(3)), int(cap.get(4))))
-            is_recording = True
-            recording_start_time = time.time()
-            print(f"Movimento detectado. Gravando vídeo: {video_filename}")
-        else:
-            recording_start_time = time.time()
-            # print(f"Movimento detectado enquanto gravava.")
+def start_recording(cap):
+    timestamp = time.strftime("%Y%m%d_%H%M%S")
+    video_filename = f"movement_{timestamp}.avi"
+    fourcc = cv2.VideoWriter_fourcc(*'XVID')
+    video_writer = cv2.VideoWriter(
+        video_filename, fourcc, 20.0, (int(cap.get(3)), int(cap.get(4))))
+    print(f"Movimento detectado. Gravando vídeo: {video_filename}")
+    return video_writer, video_filename
 
-    if is_recording:
-        # Gravar o frame no vídeo
-        video_writer.write(frame)
+def main():
+    cap = initialize_video_capture()
+    is_recording = False
+    recording_start_time = 0
+    video_writer = None
+    video_filename = None
+    prev_frame = None
 
-        # Verificar se 15 segundos se passaram
-        if time.time() - recording_start_time > VIDEO_DURATION:
-            is_recording = False
-            video_writer.release()
-            print(f"Gravação concluída: {video_filename}")
+    while True:
+        ret, frame = cap.read()
+        if not ret:
+            break
 
-    # Mostrar o vídeo ao vivo
-    cv2.imshow('Video', frame)
+        gray = process_frame(frame)
 
-    # Atualizar o frame anterior
-    prev_frame = gray
+        if prev_frame is None:
+            prev_frame = gray
+            continue
 
-    # Fechar o vídeo ao pressionar 'q'
-    if cv2.waitKey(1) & 0xFF == ord('q'):
-        break
+        contours = detect_movement(prev_frame, gray)
+        movement_detected = draw_contours(frame, contours)
 
-cap.release()
-if video_writer is not None:
-    video_writer.release()
-cv2.destroyAllWindows()
+        if movement_detected:
+            if not is_recording:
+                video_writer, video_filename = start_recording(cap)
+                is_recording = True
+                recording_start_time = time.time()
+            else:
+                recording_start_time = time.time()
+
+        if is_recording:
+            video_writer.write(frame)
+            if time.time() - recording_start_time > VIDEO_DURATION:
+                is_recording = False
+                video_writer.release()
+                print(f"Gravação concluída: {video_filename}")
+
+        cv2.imshow('Video', frame)
+        prev_frame = gray
+
+        if cv2.waitKey(1) & 0xFF == ord('q'):
+            break
+
+    cap.release()
+    if video_writer is not None:
+        video_writer.release()
+    cv2.destroyAllWindows()
+
+if __name__ == "__main__":
+    main()

